Tests/Unit/NetworkServerTests/ModuleConnectionHostTest.cs

```diff
@@ -0,0 +1,256 @@
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Shared;
+    using Moq;
+    using System;
+    using System.Configuration;
+    using System.Net;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+
+    public class ModuleConnectionHostTest
+    {
+        private readonly Faker faker = new Faker();
+
+        [Fact]
+        public void When_Constructor_Receives_Null_Parameters_Should_Throw()
+        {
+            var networkServerConfiguration = new NetworkServerConfiguration();
+            var classCMessageSender = Mock.Of<IClassCDeviceMessageSender>();
+            var loRaDeviceRegistry = Mock.Of<ILoRaDeviceRegistry>();
+
+            // ASSERT
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(null, classCMessageSender, loRaDeviceRegistry));
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, null, loRaDeviceRegistry));
+            Assert.Throws<ArgumentNullException>(() => new ModuleConnectionHost(networkServerConfiguration, classCMessageSender, null));
+        }
+
+        [Fact]
+        public async Task On_Desired_Properties_Correct_Update_Should_Update()
+        {
+            var networkServerConfiguration = new NetworkServerConfiguration();
+            var classCMessageSender = Mock.Of<IClassCDeviceMessageSender>();
+            var loRaDeviceRegistry = Mock.Of<ILoRaDeviceRegistry>();
+            var loRaModuleClient = new Mock<ILoraModuleClient>();
+            loRaModuleClient.Setup(x => x.DisposeAsync());
+
+            await using var moduleClient = new ModuleConnectionHost(networkServerConfiguration, classCMessageSender, loRaDeviceRegistry);
+            var url1 = this.faker.Internet.Url();
+            var input = $"{{\"FacadeServerUrl\": \"{url1}\"}}";
+            await moduleClient.OnDesiredPropertiesUpdate(new Microsoft.Azure.Devices.Shared.TwinCollection(input), null);
+            Assert.Equal(url1 + "/", networkServerConfiguration.FacadeServerUrl.ToString());
+            var authCode = this.faker.Internet.Password();
+            var url2 = this.faker.Internet.Url();
+            var input2 = $"{{\"FacadeServerUrl\": \"{url2}\", \"FacadeAuthCode\":\"{authCode}\"}}";
+            await moduleClient.OnDesiredPropertiesUpdate(new Microsoft.Azure.Devices.Shared.TwinCollection(input2), null);
+            Assert.Equal(url2 + "/", networkServerConfiguration.FacadeServerUrl.ToString());
+            Assert.Equal(authCode, networkServerConfiguration.FacadeAuthCode.ToString());
+
+        }
+
+        [Theory]
+        [InlineData("{\"FacadeServerUrl\": \"url2\", \"FacadeAuthCode\":\"authCode\"}")]// not a url
+        [InlineData("{\"FacadeAuthCode\":\"authCode\"}")] // no Url
+        [InlineData("{\"FacadeServerUrl\": \"\", \"FacadeAuthCode\":\"authCode\"}")]// empty url
+
```

Comment:
Consider making these and other JSON literals a little easier on the eyes by using unquoted fields and single-quoted string values:

```suggestion
        [InlineData("{ FacadeServerUrl: 'url2', FacadeAuthCode: 'authCode' }")]// not a url
        [InlineData("{ FacadeAuthCode: 'authCode' }")] // no Url
        [InlineData("{ FacadeServerUrl: '', FacadeAuthCode: 'authCode' }")]// empty url
```

This works because Json.NET permits a more relaxed JSON syntax.

Reply:
Did not know that this is very cool